src/uriparser.jl

```diff
@@ -1,5 +1,7 @@
 using .URIParser
 
+Base.depwarn("`URIParser` is deprecated, use `URIs` instead.", :URIParser, force=true)
```

Comment:
Yep, this is pretty much what I was thinking. I guess we could also move it into the constructor itself if we wanted to avoid being noisy, but I don't think it's a big deal either way 👍🏻